src/couch_mrview/src/couch_mrview.erl

```diff
@@ -531,17 +541,25 @@ map_fold({{Key, Id}, Val}, _Offset, Acc) ->
         user_acc=UAcc1,
         last_go=Go
     }};
-map_fold({<<"_local/",_/binary>> = DocId, {Rev0, Body}}, _Offset, #mracc{} = Acc) ->
+map_fold(#doc{id = <<"_local/", _/binary>>} = Doc, _Offset, #mracc{} = Acc) ->
```

Comment:
I had to stop here for quite some time. Leaving comment for future reviewers. This change would work because we open local btree differently. We pass split and join callbacks [to couch_btree:open for local_tree]( https://github.com/apache/couchdb/blob/a41c3b12896e027b1b154139f2c164551b4c5b2b/src/couch/src/couch_bt_engine.erl#L695:L696) now.